src/stores/widgets/StopGapWidget.ts

```diff
@@ -521,8 +526,12 @@ export class StopGapWidget extends EventEmitter {
             }
 
             if (isBeforeMark) {
-                // Ignore the event: it is before our interest.
-                return;
+                isRelationToUnknown =
+                    !ev.replyEventId && !!ev.relationEventId && !room.findEventById(ev.relationEventId);
+                if (!isRelationToUnknown) {
+                    // Ignore the event: it is before our interest.
+                    return;
+                }
```

Comment:
I think this seems reasonable, although maybe we should default to sending the event on if we can't find either the event or the receipt, rather than special-casing events whose parent we can't find? (ie. rename `isBeforeMark ` to `shouldForward` or something, set it to true to start with and set it to false in the first clause rather than the second... and comment it, obviously).

Reply:
I have changed `isBeforeMark` to `shouldForward` that makes easier to understand this logic, but I can’t follow on the suggestion to change:

> although maybe we should default to sending the event on if we can't find either the event or the receipt, rather than special-casing events whose parent we can't find 

It not looks clear to me how event or the receipt find could resolve the same issue. Could you please explain more exactly what you mean with that?

Reply:
@dbkr could you please have a look on the changes?

Reply:
Okay, I was just suggesting that, since the problem here is that we can't prove that the event is after the read marker because we can't find the event, we could also just default to forwarding if we can't prove the event hasn't been read, which I think would probably be safe enough. That said, this is a smaller change, so it seems fine.